substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/monitor/MultiThreadedMonitorSupport.java

```diff
@@ -192,8 +193,10 @@ public class MultiThreadedMonitorSupport extends MonitorSupport {
      * Secondary storage for monitor slots. Synchronized to prevent concurrent access and
      * modification.
      */
-    private final Map<Object, ReentrantLock> additionalMonitors = new WeakIdentityHashMap<>();
+    private final Map<Object, OwnedReentrantLock> additionalMonitors = new WeakIdentityHashMap<>();
+    private final Map<Object, Long> monitorOwners = new WeakIdentityHashMap<>();
```

Comment:
The fields `monitorOwners` and `monitorOwnersLock` are unused.

Reply:
removed!